src/fsharp/FSharp.Core/string.fs

```diff
@@ -180,3 +180,52 @@ namespace Microsoft.FSharp.Core
             else
                 let rec check i = (i < str.Length) && (predicate str.[i] || check (i+1)) 
                 check 0  
+        
+        [<CompiledName("Contains")>]
+        let contains (substring: string) (str: string) =
+            if isNull str then
+                nullArg (nameof(str))
+            else
+                str.Contains substring
+
+        [<CompiledName("Replace")>]
+        let replace (oldValue: string) (newValue: string) (str: string) =
+            if isNull str then
+                nullArg (nameof(str))
+            else
+                str.Replace(oldValue, newValue)
+
+        [<CompiledName("ReplaceChar")>]
+        let replaceChar (oldValue: char) (newValue: char) (str: string) =
+            if isNull str then
+                nullArg (nameof(str))
+            else
+                str.Replace(oldValue, newValue)
+                
+        [<CompiledName("Trim")>]
+        let trim (str: string) =
+            if isNull str then
+                nullArg (nameof(str))
+            else
+                str.Trim ()
+                
+        [<CompiledName("TrimChars")>]
+        let trimChars (chars: seq<char>) (str: string) =
+            if isNull str then
+                nullArg (nameof(str))
+            else
+                str.Trim (Seq.toArray chars)
```

Comment:
Maybe something like that could be done to avoid creating new array enumerator?

```
match chars with
| :? array<char> as chars ->
  str.Trim(chars)
 | _ ->
  str.Trim(Seq.toArray chars)
```

Reply:
Good idea. Done.